src/ol/interaction/drawboxinteraction.js

```diff
@@ -0,0 +1,360 @@
+ * @return {boolean} Stop drag sequence?
+ * @this {ol.interaction.DrawBox}
+ * @private
+ */
+ol.interaction.DrawBox.handleUpEvent_ = function(event) {
+  if (this.startCoordinate_) {
+    this.finishDrawing();
+  } else {
+    this.startDrawing_(event);
+  }
+  return false;
+};
+
+
+/**
+ * Handle move events.
+ * @param {ol.MapBrowserEvent} event A move event.
+ * @return {boolean} Pass the event to other interactions.
+ * @private
+ */
+ol.interaction.DrawBox.prototype.handlePointerMove_ = function(event) {
+  if (!goog.isNull(this.startCoordinate_)) {
+    this.modifyDrawing_(event);
+  } else {
+    this.createOrUpdateSketchPoint_(event);
+  }
+
+  return true;
+};
+
+
+/**
+ * @param {ol.MapBrowserEvent} event Event.
+ * @private
+ */
+ol.interaction.DrawBox.prototype.createOrUpdateSketchPoint_ = function(event) {
+  var coordinates = event.coordinate.slice();
+  if (goog.isNull(this.sketchPoint_)) {
+    this.sketchPoint_ = new ol.Feature(new ol.geom.Point(coordinates));
+    this.updateSketchFeatures_();
+  } else {
+    var sketchPointGeom = this.sketchPoint_.getGeometry();
+    goog.asserts.assertInstanceof(sketchPointGeom, ol.geom.Point);
+    sketchPointGeom.setCoordinates(coordinates);
+  }
+};
+
+
+/**
+ * Start the drawing.
+ * @param {ol.MapBrowserEvent} event Event.
+ * @private
+ */
+ol.interaction.DrawBox.prototype.startDrawing_ = function(event) {
+  var start = event.coordinate;
+  this.startCoordinate_ = start;
+  this.sketchLine_ = new ol.Feature(new ol.geom.LineString([start.slice(),
+        start.slice()]));
+  var geometry = new ol.geom.Polygon([[start.slice(), start.slice()]]);
+  this.sketchFeature_ = new ol.Feature();
+  if (goog.isDef(this.geometryName_)) {
+    this.sketchFeature_.setGeometryName(this.geometryName_);
+  }
+  this.sketchFeature_.setGeometry(geometry);
+
+  this.updateSketchFeatures_();
+  this.dispatchEvent(new ol.DrawEvent(ol.DrawEventType.DRAWSTART,
+      this.sketchFeature_));
+};
+
+
+/**
+ * Modify the drawing.
+ * @param {ol.MapBrowserEvent} event Event.
+ * @private
+ */
+ol.interaction.DrawBox.prototype.modifyDrawing_ = function(event) {
+  var newCoord = event.coordinate.slice();
+  var geometry = this.sketchFeature_.getGeometry();
+  var lsGeom = this.sketchLine_.getGeometry();
+  var pointGeom = this.sketchPoint_.getGeometry();
+
+  goog.asserts.assertInstanceof(geometry, ol.geom.Polygon);
+  goog.asserts.assertInstanceof(lsGeom, ol.geom.LineString);
+  goog.asserts.assertInstanceof(pointGeom, ol.geom.Point);
+
+  var start = this.startCoordinate_;
+
+  var width = newCoord[0] - start[0];
+  var height = newCoord[1] - start[1];
+
+  var first = [start[0], start[1]];
+  var second = [start[0], start[1]];
+
+  if (this.condition_(event)) {
+    var maxLen = Math.max(Math.abs(width), Math.abs(height));
+
+    if (width > 0) { second[0] += maxLen; } else { first[0] -= maxLen; }
+    if (height > 0) { second[1] += maxLen; } else { first[1] -= maxLen; }
+  }
+  else
+  {
```

Comment:
For consistency with if/else blocks elsewhere, this should be `} else {` (on one line).